cmd/sink/logs/logs.go

```diff
@@ -7,91 +7,80 @@ import (
 	"context"
 	"fmt"
 	"log/slog"
+	"os"
 	"strings"
 
 	ocel "github.com/kubearchive/kubearchive/pkg/cel"
 	"github.com/kubearchive/kubearchive/pkg/logurls"
 	"github.com/kubearchive/kubearchive/pkg/models"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	celPrefix        = "cel:"
 	containerNameCel = "cel:spec.containers.map(m, m.name)"
+	loggingDirEnvVar = "KUBEARCHIVE_LOGGING_DIR"
 )
 
-var (
-	configCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Name:      "kubearchive-config",
-		Namespace: "kubearchive",
-	}
-	loggingCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Namespace: "kubearchive",
+func filesInDir(path string) (map[string]string, error) {
+	files := make(map[string]string)
+	dir, err := os.ReadDir(path)
+	if err != nil {
+		return files, err
 	}
-	loggingCmSelector = corev1.ConfigMapKeySelector{
-		Key: "logging-configmap-name",
+	for _, file := range dir {
+		// kubernetes will create symlinks that start with '..' that should be ignored
+		if !file.IsDir() && !strings.HasPrefix(file.Name(), "..") {
+			var filePath strings.Builder
+			fmt.Fprintf(&filePath, "%s/%s", path, file.Name())
+			files[file.Name()] = filePath.String()
+		}
 	}
-)
+	return files, nil
+}
 
-func getKubeArchiveLoggingCm(ctx context.Context, kubeClient kubernetes.Interface) *corev1.ConfigMap {
-	emptyCm := &corev1.ConfigMap{Data: map[string]string{}}
-	kaConfigCm, err := kubeClient.CoreV1().ConfigMaps(configCmRef.Namespace).Get(
-		ctx,
-		configCmRef.Name,
-		metav1.GetOptions{},
-	)
-	if err != nil {
-		slog.Warn(
-			"error while getting ConfigMap",
-			"ConfigMap",
-			fmt.Sprintf("%s/%s", configCmRef.Namespace, configCmRef.Name),
-			"error",
-			err.Error(),
-		)
-		return emptyCm
+func loggingConfigFromFiles(files map[string]string) (map[string]string, error) {
```

Comment:
Can we move this logic to `pkg`? With the recent changes on the logging proposal, the api will also need to load this configmap information

Reply:
I moved `LoggingConfigFromFiles` and `FilesInDir` into the `pkg/files`